### DISCUSSION TITLE
Unable to use Trainer.predict: 'Trainer' object has no attribute 'predict_dataloaders'

### DISCUSSION THREAD

**Discussion by its-dron** [Lightning-AI/pytorch-lightning#6367] [General]
## 🐛 Bug

Unable to use the new `Trainer.predict` method with datamodule input. The following error is thrown: `AttributeError: 'Trainer' object has no attribute 'predict_dataloaders'`

Best I can tell, the error originates from this line: https://github.com/PyTorchLightning/pytorch-lightning/blob/6cc1a060785448eacd016f6689d592ac822adaaf/pytorch_lightning/trainer/predict_loop.py#L33

Additionally, the documentation is vague about which dataloader is used from the datamodule.

### To Reproduce

https://colab.research.google.com/drive/1ZD5xwVifXnxkbAypyeP-nnAXa1IYCBNx?usp=sharing

### Expected behavior

Don't throw an error?

### Environment

```
* CUDA:
	- GPU:
		- Tesla T4
	- available:         True
	- version:           10.1
* Packages:
	- numpy:             1.19.5
	- pyTorch_debug:     False
	- pyTorch_version:   1.7.1+cu101
	- pytorch-lightning: 1.2.2
	- tqdm:              4.41.1
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.7.10
	- version:           #1 SMP Thu Jul 23 08:00:38 PDT 2020
```

 - PyTorch Version (e.g., 1.0):
 - OS (e.g., Linux):
 - How you installed PyTorch (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

**Comment by tchaton** [comment]
Dear @its-dron,

It is still an experimental feature and will change. It is why it hasn t been added to the doc yet. But you can add a predict_dataloader to your DataModule and it should work. It will call the predict function of the Lightning Module you can override. This will likely be renamed predict_step soon

Best,
T.C
